### PR TITLE
SNOW-414875 Add GET documentation for Golang

### PR DESCRIPTION
**Pull request by sfc-gh-sleslie**
SNOW-414875
Summary of changes:
* Adding documentation about support for GET command.
* Minor revisions to the PUT documentation

Testing: local HTML build to check presentation and clarity
y

### Description
Please explain the changes you made here.

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sfc-gh-jbahk — APPROVED** [review]
LGTM

File: doc.go [hunk 1]
```diff
@@ -267,10 +267,10 @@ data types. The columns are:
     INTEGER that         | int, int8, int16, int32, int64              | string                 | int, int8, int16,
     fits in int64        |                                   [1] , [2] |                        | int32, int64
   -------------------------------------------------------------------------------------------------------------------
-    INTEGER that doesn't | int, int8, int16, int32, int64, \*big.Int   | string                 | error
+    INTEGER that doesn't | int, int8, int16, int32, int64,  *big.Int   | string                 | error
```
Comment (sfc-gh-sleslie) [inline doc.go:270]: @sfc-gh-jbahk Mark and I discussed this, but can you verify that the slash is not required.

File: doc.go [hunk 2]
```diff
@@ -297,7 +297,7 @@ data types. The columns are:
   [2] Attempting to convert from a higher precision data type to a lower precision data type via interface{}
       causes an error.
 
-  [3] Higher precision data types like \*big.Int and \*big.Float can be accessed by querying with a context
```
Comment (sfc-gh-sleslie) [inline doc.go:300]: @sfc-gh-jbahk Mark and I discussed this, but can you verify that the slash is not required.

**Review by sfc-gh-mgilkey — APPROVED** [review]
LGTM

File: doc.go (hunk 1 above)
Comment (sfc-gh-jbahk) [reply doc.go:270]: The \ is not required in real text but I'm unsure how it would appear in the doc. Feel free to delete if there's no impact on the doc.

Comment (sfc-gh-sleslie) [reply doc.go:270]: Thanks! It looked fine in my local build using godoc.

**Pull request merged**